finetuner/finetuner.py

```diff
@@ -0,0 +1,181 @@
+import os
+from typing import List, Optional
+
+import hubble
+from finetuner.client import FinetunerV1Client
+from finetuner.constants import CREATED_AT, DESCRIPTION, NAME, STATUS
+from finetuner.experiment import Experiment
+from finetuner.names import get_random_name
+from finetuner.run import Run
+
+
+class Finetuner:
+    """Finetuner class."""
+
+    def __init__(self):
+        self._client = None
+        self._default_experiment = None
+
+    def login(self):
+        """Login to Hubble account, initialize a client object
+        and create a default experiment.
+
+        Note: Calling `login` is necessary for using finetuner.
+        """
+        hubble.login()
+        self._client = FinetunerV1Client()
+        self._default_experiment = self._get_default_experiment()
+
+    def _get_default_experiment(self) -> Experiment:
+        """Create or retrieve (if it already exists) a default experiment
+        for the current working directory."""
+        experiment_name = os.getcwd().split('/')[-1]
+        for experiment in self.list_experiments():
+            if experiment.get_name() == experiment_name:
+                return experiment
+        experiment = self.create_experiment(name=experiment_name)
+        return experiment
+
+    def create_experiment(self, name: Optional[str] = None) -> Experiment:
+        """Create an experiment.
+
+        :param: Optional name of the experiment.
+        :returns: An `Experiment` object.
+        """
+        if not name:
+            name = get_random_name()
+        experiment_info = self._client.create_experiment(name=name)
+        experiment = Experiment(
+            client=self._client,
+            name=experiment_info[NAME],
+            status=experiment_info[STATUS],
+            created_at=experiment_info[CREATED_AT],
+            description=experiment_info[DESCRIPTION],
+        )
+        return experiment
+
+    def get_experiment(self, name: str) -> Experiment:
+        """Get an experiment by its name.
+
+        :param name: Name of the experiment.
+        :returns: An `Experiment` object.
+        """
+        experiment_info = self._client.get_experiment(name=name)
+        experiment = Experiment(name=experiment_info[NAME], client=self._client)
+        return experiment
+
+    def list_experiments(self) -> List[Experiment]:
+        """List every experiment."""
+        experiment_infos = self._client.list_experiments()
+        experiments = [
```

Comment:
instead of creating a variable just to return it, you can do it in-place:
```
return [
            Experiment(
                client=self._client,
                name=experiment_info[NAME],
                status=experiment_info[STATUS],
                created_at=experiment_info[CREATED_AT],
            )
            for experiment_info in experiment_infos
        ]
```
I think it looks cleaner